rclpy/test/test_parameter_client.py

```diff
@@ -0,0 +1,159 @@
+# Copyright 2022 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from tempfile import NamedTemporaryFile
+import unittest
+
+import rcl_interfaces.msg
+from rcl_interfaces.msg import ParameterType
+import rcl_interfaces.srv
+import rclpy
+import rclpy.context
+from rclpy.executors import SingleThreadedExecutor
+from rclpy.parameter import Parameter
+from rclpy.parameter_client import AsyncParameterClient
+
+
+class TestParameterClient(unittest.TestCase):
+
+    def setUp(self):
+        self.context = rclpy.context.Context()
+        rclpy.init(context=self.context)
+        self.client_node = rclpy.create_node('test_parameter_client',
+                                             namespace='/rclpy',
+                                             context=self.context)
+        self.target_node = rclpy.create_node('test_parameter_client_target',
+                                             namespace='rclpy',
+                                             allow_undeclared_parameters=True,
+                                             context=self.context)
```

Comment:
style nitpick: our [style guide](https://docs.ros.org/en/rolling/The-ROS2-Project/Contributing/Code-Style-Language-Versions.html#python) prefers hanging indents, breaking directly after the opening parenthesis

```suggestion
        self.client_node = rclpy.create_node(
            'test_parameter_client',
            namespace='/rclpy',
            context=self.context)
        self.target_node = rclpy.create_node(
            'test_parameter_client_target',
            namespace='rclpy',
            allow_undeclared_parameters=True,
            context=self.context)
```

unfortunately, I don't think our linters check for this :shrug: